### PR TITLE
ability to overload 'make' in a shell session

### LINKED ISSUE #179: [PRIO] make inside docker
scenario: a repo designates a docker image to be its sandbox for building software. Every `make` command should be executed inside a docker container with the repo (current folder with a `Makefile`) as a mounted volume.

`make foo` basically needs to be intercepted, and rather than invoke it locally, it needs to be invoked inside the docker container

alternative 1: why intercept when you can just start a shell inside the docker container? because that will be a new environment altogether. It will create confusion and frustration since the developer expects to develop on their own machine, in their own customized environment.

alternative 2: why intercept when you can just create another command e.g. `dake` (japanese for rain cloud, fog, mist, haze; or see Urban dictionary https://www.urbandictionary.com/define.php?term=Dake) which would execute `make $@` inside the docker container? because it's a matter of time before someone types `make build` and since you're mounting the repo as a volume, you end up compiling for your own architecture, instead of the one inside the docker container.

But why mount in the first place? Because it's easier and faster to make sure permissions are ok (UID and GID) than to copy the entire folder (including `.git` which might be big). You'd also miss out on the ability to sync quickly e.g. edit locally and run tests inside docker.

alternative 3: would be great if GNU make itself upon execution could check if it needs to run inside a docker container, and execute same invocation there, and exit early without invoking the targets locally as well. It doesn't seem possible to do that without `$(error ...)` which would always have a failed exit code.

**alternative 4 - LEADER for now**: intercept. when sourcing `dev.inc.sh`, one could also source a shell function called `make`. This function, if it detects a marker or actual configuration for running `make` inside a docker container, upon invocation it would do so. If it doesn't detect that, it would invoke the original `make` command based on `which -a make` (second item).

### PR DESCRIPTION
**Pull request by andreineculau**
* Fixes: #179 
* Breaking change: [ ]

---

This allows for various useful scenarios, like:
* run all `make` commands on a development inside a sandbox (docker container) as described in #179 
* implement different make behaviour in a CI environment
* implement different make behaviour inside a sandbox (docker container)
* ?

I don't see any read negatives (and that's a problem! :) ), except for this corner-case of already having `make` as an alias/function.

cc @erpheus

### CODE REVIEW

File: sh/exe-env.inc.sh [hunk 1]
```diff
@@ -1,45 +1,62 @@
...
-    for f in coreutils findutils gnu-sed gnu-tar gnu-time gnu-which grep gzip make; do
-        path_prepend ${HOMEBREW_PREFIX}/opt/${f}/libexec/gnubin
-    done
-    path_prepend ${HOMEBREW_PREFIX}/opt/curl/bin
-    path_prepend ${HOMEBREW_PREFIX}/opt/gettext/bin
-    path_prepend ${HOMEBREW_PREFIX}/opt/unzip/bin
-    path_prepend ${HOMEBREW_PREFIX}/opt/zip/bin
-
-    unset HOMEBREW_PREFIX
+(
```
Comment (erpheus) [inline sh/exe-env.inc.sh]: Just wondering, why was this moved to a subshell?

**Review by erpheus — APPROVED** [review]
LGTM

I like the `SF_MAKE_PASS` and `SF_MAKE_SH_PASS` variables 👌

**Comment by andreineculau** [comment]
> I like the SF_MAKE_PASS and SF_MAKE_SH_PASS variables

fixed the typo. Thanks!

File: sh/exe-env.inc.sh (hunk 1 above)
Comment (andreineculau) [reply sh/exe-env.inc.sh]: unintentional

**Comment by erpheus** [comment]
> fixed the typo. Thanks!

I thought they were supposed to be complementary 🤦
